dali/python/nvidia/dali/plugin/jax/clu.py

```diff
@@ -221,3 +226,105 @@ def element_spec(self) -> ElementSpec:
             ElementSpec: Element spec for the elements returned by the iterator.
         """
         return self._element_spec
+
+
+def peekable_data_iterator(
+        pipeline_fn=None,
+        output_map=[],
+        size=-1,
+        reader_name=None,
+        auto_reset=False,
+        last_batch_padded=False,
+        last_batch_policy=LastBatchPolicy.FILL,
+        prepare_first_batch=True,
+        sharding=None):
+    """Decorator for DALI pipelines that returns a peekable iterator. Compatible with Google CLU
+    PeekableIterator. It supports peeking the next element in the iterator without advancing the
+    iterator.
+
+    Parameters
+    ----------
+    pipeline_fn function:
+                Function to be decorated. It should be comaptible with
+                :meth:`nvidia.dali.pipeline.pipeline_def` decorator.
+                For multigpu support it should accept `device_id`, `shard_id` and `num_shards` args.
+    output_map : list of str
+                List of strings which maps consecutive outputs
+                of DALI pipelines to user specified name.
+                Outputs will be returned from iterator as dictionary
+                of those names.
+                Each name should be distinct
+    size : int, default = -1
+                Number of samples in the shard for the wrapped pipeline (if there is more than
+                one it is a sum)
+                Providing -1 means that the iterator will work until StopIteration is raised
+                from the inside of iter_setup(). The options `last_batch_policy` and
+                `last_batch_padded` don't work in such case. It works with only one pipeline inside
+                the iterator.
+                Mutually exclusive with `reader_name` argument
+    reader_name : str, default = None
+                Name of the reader which will be queried to the shard size, number of shards and
```

Comment:
```suggestion
                Name of the reader which will be queried for the shard size, number of shards and
```
?

Reply:
Done for all iterators since this was copied between them.